service/history/workflow/retry.go

```diff
@@ -63,33 +100,25 @@ func getBackoffInterval(
 	// (we made 1 non-retry attempt already) and condition (currentAttemptCounterValue+1 > maxAttempts) will be false.
 	// With 2nd retry, currentAttemptCounterValue will be 2 (1 non-retry + 1 retry attempt already made) and
 	// condition (currentAttemptCounterValue+1 > maxAttempts) will be true (means stop retrying, we tried 2 times already).
-	if maxAttempts > 0 && currentAttemptCounterValue+1 > maxAttempts {
+	if maxAttempts > 0 && currentAttempt+1 > maxAttempts {
```

Comment:
Please rename `currentAttemptCounterValue` in the comment above also.

Reply:
I think this is better:
```suggestion
	if maxAttempts > 0 && currentAttempt >= maxAttempts {
```